ringbuf/reader_test.go

```diff
@@ -234,10 +240,68 @@ func TestReaderBlocking(t *testing.T) {
 	}
 }
 
+func TestReaderNoWakeup(t *testing.T) {
+	testutils.SkipOnOldKernel(t, "5.8", "BPF ring buffer")
+
+	prog, events := mustOutputSamplesProg(t,
+		sampleMessage{size: 5, flags: unix.BPF_RB_NO_WAKEUP}, // Read after timeout
+		sampleMessage{size: 6, flags: unix.BPF_RB_NO_WAKEUP}, // Discard
+		sampleMessage{size: 7, flags: unix.BPF_RB_NO_WAKEUP}) // Read won't block
+
+	rd, err := NewReader(events)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer rd.Close()
+
+	ret, _, err := prog.Test(internal.EmptyBPFContext)
+	testutils.SkipIfNotSupported(t, err)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if errno := syscall.Errno(-int32(ret)); errno != 0 {
+		t.Fatal("Expected 0 as return value, got", errno)
+	}
+
+	var timeoutMs int64 = 100
+	lowerBound := timeoutMs * 98 / 100
+	upperBound := timeoutMs * 102 / 100
+	nonBlocking := timeoutMs * 2 / 100
+	startTime := time.Now()
+
+	rd.SetDeadline(startTime.Add(time.Duration(timeoutMs) * time.Millisecond))
```

Comment:
This could just be
```suggestion
	rd.SetDeadline(time.Now())
```

This way we do a non-blocking read and can then check whether it returned a sample.